core/models.py

```diff
@@ -143,6 +157,10 @@ class EventPage(models.Model):
     url = models.CharField(max_length=200, null=True, blank=True)
 
     is_live = models.BooleanField(null=False, blank=False, default=False)
+    is_deleted = models.BooleanField(default=False)
+
+    objects = EventPageManager()
+    all_objects = models.Manager()  # This includes deleted objects
```

Comment:
Could we call this `with_deleted`? Otherwise we've got EventPage.objects.all() and EventPage.all_objects.all(), which is a bit weird :)